### DISCUSSION TITLE
Resolution - Keep Searching Until Resolution Found

### DISCUSSION THREAD

**Discussion by godver3** [itsToggle/plex_debrid#130] [Q&A]
Hi - I was wondering if it is possible to set this up so that it searches for both 2160p and 1080p releases, and to continue searching even if one is found? The issue that I am experiencing is that the tool discovers a 1080p release, but due to tracker error does not discover a 2160p release (even though one exists), and then considers the search "complete" because it found a 1080p release. Not sure if this makes sense, or if I am missing something. I am currently set up through Jackett and perhaps don't have it set up correctly, but noticing frequently that 2160p releases aren't pulled at the same time as 1080p. Thanks!

**Comment by itsToggle** [comment]
Hi, am I right in assuming that you've already set up the "Multiple Versions" setting to download both qualities? Would you mind telling me the string you used for the versions you've defined?

I have added this as a new issue and marked it as an enhancement, because I'm very much interested in this feature aswell. What Im noticing for example, is that right after a new episode comes out, 2160p releases take longer to be released than 1080p releases. Plex_debrid will therefore only download 1080p releases for this newly released episode and never try to download 2160p releases again. I will write a process that regularly checks if better qualities are available :)

**Reply by godver3** [reply]
Hi there - this is what I have entered:

![88BD04ED-4D95-452A-B626-47B336C8F0A9](https://user-images.githubusercontent.com/33110414/191331931-690619b5-e821-4672-a18b-f32bbab02f3f.png)

I ran some searches this morning and in general it successfully snatches both, but just like you’ve described if the timing is off it will give up after finding one. One option I thought about was to run two instances of the script, and sort into two libraries. This could only be done if there were multiple folders within the RD rclone (i.e. movies-4k, movies-1080p etc), otherwise a separate RD account would be needed. 

Thanks for your responsiveness - I imagine a future where this software joins the “arr” family - it does seem like the goal is to essentially replicate much of that functionality. To be honest it is already 95% there and covers most use cases.

**Reply by itsToggle** [reply]
There is a very long way to go to join the *arr family, as Im lacking the programming knowledge and time to perfect this script. 
The *arr family has a very high standard. I might try to implement my project as a plugin for Sonarr and radarr in the near future, which would be a more realistic goal ;)

The versions you are using look fine! If you have a sufficient amount of indexers added to jackett I don't see why the script would have a problem adding both qualities. I'll look through the code again!

**Reply by godver3** [reply]
No worries - I suspect in this case it relates to the indexer. I think separate folders by resolution would be a workaround for now that would allow multiple instances of Plex_Debrid to run and have each set to a different resolution.

**Comment by godver3** [comment]
Hi there - this is what I have entered:


I ran some searches this morning and in general it successfully snatches
both, but just like you’ve described if the timing is off it will give up
after finding one. One option I thought about was to run two instances of
the script, and sort into two libraries. This could only be done if there
were multiple folders within the RD rclone (i.e. movies-4k, movies-1080p
etc), otherwise a separate RD account would be needed.

Thanks for your responsiveness - I imagine a future where this software
joins the “arr” family - it does seem like the goal is to essentially
replicate much of that functionality. To be honest it is already 95% there
and covers most use cases.

**Reply by itsToggle** [reply]
Hey! Don't know if you are still typing or not ;) This issue should be resolved since plex_debrid v1.30 (yesterday). The "multiple versions" and "release sorting" settings have now been combined into one setting called "versions". This new settings allows plex_debrid to keep searching until all versions have been downloaded :)

**Comment by mrkrauz123** [comment]
Hi, I don't mean to hijack this discussion but my question is kinda related so I am going to ask it here. I am currently using the following filters for my multiple versions:

  ```
"Multiple versions": [
        [
            "(.HDR.)"
        ],
        [
            "!(.HDR.)"
        ],
        [
            "(2160p)"
        ],
        [
            "(1080p)"
        ],
        [
            "(720p)"
        ]
```

This is an extract from my settings file. Mostly it works great but sometimes when searching for the non-hdr version instead of getting a 4k non-hdr file it gets a 1080p non-hdr file. I obviously get why that is and it makes sense that the script works that way but I was wondering if there is an option to combine two filters into one so that they are dependent of each other? What I mean is this:
I would love the script to search for a 4k HDR file, a 4k non-HDR file, a 1080p HDR file (This one isn't as common but sometimes it exists) and lastly a 1080p non-HDR file. How do I tell the script to only add a release if it meets both of my requirements ("(.HDR.)" AND "(2160p)" for example). I hope you know what I mean and that I didn't explain it too poorly. 
Thanks in advance for your help and sorry again if this question doesn't belong here, though I hope that it's fine.

**Reply by itsToggle** [reply]
Hi, plex_debrid has been updated quite a lot lately. The multiple versions setting has been completely rewritten! It should be capable of what you mention here.

**Reply by mrkrauz123** [reply]
Oh wow, I just downloaded the latest version and it's amazing. Really great things you've done.

**Reply by mrkrauz123** [reply]
Will it still submit releases to RD whenever no cached ones are available for a given resolution? I was just looking through the settings menu and couldn't find that option to enable it, which is why I am asking.

**Reply by itsToggle** [reply]
Thanks! :) You can now define that for each version. In both default versions there is an attribute called "cache status" which by default is "required" to be "cached". To allow the download of uncached releases for a specific version, simply change the cache status rule to be a "preference", rather than a "requirement" to be cached.
